File: remove_bom.py
import glob
import os

html_files = glob.glob('*.html')
count = 0

for file_path in html_files:
    try:
        # Leer como bytes para detectar el BOM
        with open(file_path, 'rb') as f:
            raw_content = f.read()

        # Si inicia con BOM UTF-8 (\xef\xbb\xbf), quitarlo
        if raw_content.startswith(b'\xef\xbb\xbf'):
            raw_content = raw_content[3:]
            
        # Decodificar y escribir explícitamente en UTF-8 puro
        text = raw_content.decode('utf-8')
        with open(file_path, 'w', encoding='utf-8', newline='') as f:
            f.write(text)
        count += 1
        print(f"[OK] {file_path} -> Limpio y sin BOM")
    except Exception as e:
        print(f"[ERROR] {file_path}: {e}")

print(f"\nFinalizado. {count} archivos procesados exitosamente.")

File: remove_bom.js
const fs = require('fs');

// Use native fs.readdir to avoid installing glob dependency if missing
fs.readdir('.', (err, files) => {
    if (err) throw err;
    
    const htmlFiles = files.filter(f => f.endsWith('.html'));
    let count = 0;
    
    for (const file of htmlFiles) {
        let buffer = fs.readFileSync(file);
        
        // Remove BOM if present
        if (buffer[0] === 0xef && buffer[1] === 0xbb && buffer[2] === 0xbf) {
            buffer = buffer.slice(3);
        }
        
        // Convert back to string and write as UTF-8 (Node writes without BOM by default)
        const text = buffer.toString('utf8');
        fs.writeFileSync(file, text, 'utf8');
        count++;
        console.log(`[OK] ${file} -> Limpio y sin BOM`);
    }
    
    console.log(`\nFinalizado. ${count} archivos procesados exitosamente.`);
});
